cli/src/utils/sourcemaps.rs

```diff
@@ -163,7 +164,17 @@ pub fn read_pairs(directory: &PathBuf) -> Result<Vec<SourcePair>> {
         if is_javascript_file(&entry_path) {
             info!("Processing file: {}", entry_path.display());
             let source = SourceFile::load(&entry_path)?;
-            let sourcemap_path = guess_sourcemap_path(&source.path);
+            let mut sourcemap_path = guess_sourcemap_path(&source.path);
+            
+            // If guess_sourcemap_path doesn't find a file, try parsing the source
+            if !sourcemap_path.exists() {
+                if let Some(sourcemap_url) = get_sourcemap_path(&source.content) {
+                    // Resolve the sourcemap URL relative to the source file
+                    let source_dir = source.path.parent().unwrap_or(Path::new("."));
+                    sourcemap_path = source_dir.join(sourcemap_url);
+                }
+            }
```

Comment:
We do a version of url extraction in https://github.com/PostHog/posthog/blob/6b8b21a7a16c2cef56022646eb559c3ba2e9fdaa/rust/cymbal/src/symbol_store/sourcemap.rs#L192-L227.

Do you have an example of somewhere the multiline case needs to be considered? I haven't come across that anywhere else before now

Reply:
I have not come across it either - this function is attempting to follow the [TC39 sourcemap standard](https://tc39.es/ecma426/#sec-JavaScriptExtractSourceMapURL) to be rigorous. In particular I have tried to translate their js example into rust. If you think the multi-line case doesn't need to be supported then it can be dropped.

I initially looked for a rust library that would do it, but couldn't find one, and also missed the existing parsing logic you have linked.